Add the capability to know which CVE’s affect an agent - Documentation changes (#3564)
wazuh/wazuh-documentation: source/user-manual/capabilities/vulnerability-detection/how_it_works.rst

Once the global vulnerability database (with the CVEs) is created, the detection process looks for vulnerable packages in the inventory databases (unique per agent). Alerts are generated when a CVE (Common Vulnerabilities and Exposures) affects a package that is known to be installed in one of the monitored servers. A package is labeled as vulnerable when its version is contained within the affected range of a CVE.
The results are presented as alerts and also stored in a database. So you can check the last scan alerts, or query every single agent's vulnerable software DB.

.. _vuln_det_scan_types:

Scan types
^^^^^^^^^^

The Vulnerability Detector module can run a scan on startup (:ref:`run_on_start <vuln_det_run_on_start>`) and every certain period of time (:ref:`interval <vuln_det_interval>`).
In any of these cases, the packages that have been already scanned will wait until the :ref:`ignore_time <vuln_det_ignore_time>` expires to be re-scanned.
This results in a better performance and we avoid a repeated alert during a configurable time. We have then two different type of scans:

> - Full scan: The first time, Vulnerability Detector scans every single package installed. From then on, all the available packages are scanned again only when the configured :ref:`ignore_time <vuln_det_ignore_time>` expires.

Reviewer: ```suggestion
- Full scan: The first time, Vulnerability Detector scans every single package installed. After this, all the available packages are scanned again only when the configured :ref:`ignore_time <vuln_det_ignore_time>` expires.
```

> - Partial scans: Only new packages are scanned, as long as :ref:`ignore_time <vuln_det_ignore_time>` is still valid.

Reviewer: ```suggestion
- Partial scans: Only new packages are scanned while :ref:`ignore_time <vuln_det_ignore_time>` is still valid.
```

There are few considerations that arise from this behavior:

- Every full scan generates alerts for all the packages, so the alerts are repeated until they get fixed.

> - The user can't trigger a full scan manually, the only option is to decrease the :ref:`ignore_time <vuln_det_ignore_time>` setting.

Reviewer: ```suggestion
- The user can't trigger a full scan manually, the only option is to decrease the :ref:`ignore_time <vuln_det_ignore_time>` setting.
```
```suggestion
- The user ca not trigger a full scan manually, the only option is to decrease the :ref:`ignore_time <vuln_det_ignore_time>` setting.
```

> - The partial scans generate alerts for new packages, but they don't delete alerts for removed packages.

Reviewer: ```suggestion
- The partial scans generate alerts for new packages, but they do not delete alerts for removed packages.
```
